mux/src/tmux_pty.rs

```diff
@@ -0,0 +1,118 @@
+use crate::{
+    tmux::{RefTmuxRemotePane, TmuxCmdQueue, TmuxDomainState},
+    tmux_commands::SendKeys,
+};
+use flume;
+use portable_pty::{Child, ExitStatus, MasterPty};
+use std::{
+    io::{Read, Write},
+    sync::{Arc, Condvar, Mutex},
+};
+
+pub(crate) struct TmuxReader {
+    rx: flume::Receiver<String>,
+}
+
+impl Read for TmuxReader {
+    fn read(&mut self, mut buf: &mut [u8]) -> std::io::Result<usize> {
+        match self.rx.recv() {
+            Ok(str) => {
+                return buf.write(str.as_bytes());
```

Comment:
This buffer reader has a potential problem: if we recv a String that is larger than buf it won't fit and we'll discard the excess.

I think we can make this safer by using a `socketpair` from our `filedescriptor` crate.  Take a look at `read_from_pane_pty` to see how it uses a socketpair as a channel for the bytes.

The tmuxreader could potentially just one of the `FileDescriptor`s from the socket pair and simplify some of the code in here.

Reply:
Can't get point of `socketpair` here, but I tried to add a buffer to avoid the "overrange" problem.